api/src/main/java/jakarta/data/StaticMetamodel.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * <p>Annotates a class to serve as a static metamodel for an entity,
+ * enabling type-safe access to the names of entity attributes.</p>
+ *
+ * <p>For example, for the following entity,</p>
+ *
+ * <pre>
+ * &#64;Entity
+ * public class Person {
+ *     &#64;Id
+ *     public long ssn;
+ *
+ *     &#64;Embedded
+ *     public Name name;
+ *
+ *     public int yearOfBirth;
+ * }
+ *
+ * &#64;Embeddable
+ * public class Name {
+ *     public String first;
+ *     public String last;
+ * }
+ * </pre>
+ *
+ * <p>You can define a static metamodel as follows,</p>
+ *
+ * <pre>
+ * &#64;StaticMetamodel(Person.class)
+ * public class Person_ {
+ *     public static volatile ssn; // or id
+ *     public static volatile name;
```

Comment:
this does not look like syntactically correct java

Reply:
> this does not look like syntactically correct java

Oops. It looks like I left out the type.  I'll add it back in.